packages/components/src/components/hds/popover-primitive/index.js

```diff
@@ -0,0 +1,245 @@
+      }
+
+      // Apply the `hds-anchored-position` modifier to the "popover" element
+      // (notice: this function runs the first time when the element the modifier was applied to is inserted into the DOM, and it autotracks while running.
+      // Any tracked values that it accesses will be tracked, including the arguments it receives, and if any of them changes, the function will run again)
+      // This modifiers uses the Floating UI library to provide:
+      // - positioning of the "popover" in relation to the "toggle"
+      // - collision detection (optional)
+      next(() => {
+        anchoredPositionModifier(
+          this.popoverElement, // element the modifier is attached to
+          [this.toggleElement], // positional arguments
+          anchoredPositionOptions // named arguments
+        );
+      });
+    },
+    { eager: true }
+  );
+
+  @action
+  showPopover() {
+    this.popoverElement.showPopover();
+  }
+
+  @action
+  hidePopover() {
+    this.popoverElement.hidePopover();
+  }
+
+  @action
+  togglePopover() {
+    this.popoverElement.togglePopover();
+  }
+
+  // fired just _before_ the "popover" is shown or hidden
+  @action
+  onBeforeTogglePopover(event) {
+    if (event.newState === 'closed') {
+      // we need this flag to check if it's in the "closing" process,
+      // because the browser automatically returns the focus to the "trigger" button
+      // and this would re-open immediately the popover because of the `focusin` event
+      this.isClosing = true;
+    }
+  }
+
+  // fired just _after_ the "popover" is shown or hidden
+  @action
+  onTogglePopover(event) {
+    if (event.newState === 'open') {
+      this.isOpen = true;
+
+      // we call the "onOpen" callback if it exists (and is a function)
+      let { onOpen } = this.args;
+      if (typeof onOpen === 'function') {
+        onOpen();
+      }
+    } else {
+      this.isOpen = false;
+
+      // reset the "isClosing" flag (the `toggle` event is fired _after_ the popover is closed)
+      this.isClosing = false;
+
+      // if the popover was initially forced to be open (using the "manual" state) then revert its status to `auto` once the user interacts with it
+      if (this.args.isOpen) {
+        this.popoverElement.popover = 'auto';
+      }
+
+      // we call the "onClose" callback if it exists (and is a function)
+      let { onClose } = this.args;
+      if (typeof onClose === 'function') {
+        onClose();
+      }
+    }
+  }
+
+  @action
+  onMouseEnter() {
+    if (this.timer) {
+      clearTimeout(this.timer);
+    }
+    this.showPopover();
+  }
+
+  @action
+  onFocusIn() {
+    // don't re-open the popover if the focus is returned because the closing
+    if (!this.isClosing) {
+      if (this.timer) {
+        clearTimeout(this.timer);
+      }
+      this.showPopover();
+    }
+  }
+
+  @action
+  onMouseLeave() {
+    this.timer = setTimeout(this.hidePopover.bind(this), 500);
```

Comment:
Not too opinionated about this, but it can also be writted as `this.timer = setTimeout(() => this.hidePopover(), 500);` depending on whether folks in our team read an arrow function easier than a bind method (I found this format to be more popular in our library)